src/librustc_typeck/diagnostics.rs

```diff
@@ -4665,6 +4625,22 @@ It is recommended that you look for a `new` function or equivalent in the
 crate's documentation.
 "##,
 
+E0642: r##"
+This error indicates that there is a mismatch between generic parameters and
+impl Trait parameters in a trait Declaration versus it's impl.
```

Comment:
Nits: "Declaration" should probably be lowercase, "s/it's/its"

Reply:
Thanks!